### ISSUE TITLE
Dialog: add option to remove 'X' button

### ISSUE THREAD

**Issue by oxala** [eBay/ebayui-core#264]
# Feature Request

## Description
I need to be able to have a dialog that user cannot close by clicking outside. It should not have an X as well.

## Context
The use-case - when we need to force user to choose (let's say Accept or Deny) something in the dialog.

**Comment by ianmcburnie** [comment]
@oxala Can you please share a real life example or situation?

**Comment by ianmcburnie** [comment]
Sounds good. We have accessibility concerns around forcing users into a potentially stressful decision they might have arrived at by mistake (e.g. after clicking the wrong button). An 'x' or tap outside allows users to defer making any such decision.

**Comment by ianmcburnie** [comment]
This should ideally be part of a new `ebay-alert-dialog` pattern or perhaps a new `type="alert"` for the current `ebay-dialog`. Alert dialogs always require a user choice and therefore the "X" is not necessary.

**Comment by ianmcburnie** [comment]
This will be tackled long term in #1164 

In the interim, for 5.x, let's just add this as a "secret" option to dialog.

**Comment by ianmcburnie** [comment]
@leksterz this could be a good first issue for you to work on. @agliga or I can assist you.

**Comment by ianmcburnie** [comment]
This will be available with the new confirm and alert dialogs in v6. Closing.
